examples/autopilot_server/autopilot_server.cpp

```diff
@@ -0,0 +1,312 @@
+        // std::future.
+        auto prom = std::make_shared<std::promise<Mission::Result>>();
+        auto future_result = prom->get_future();
+        Mission::MissionPlan mission_plan{};
+        mission_plan.mission_items = mission_items;
+        std::cout << "SystemID" << unsigned(system->get_system_id()) << std::endl;
+        mission.upload_mission_async(
+            mission_plan, [prom](Mission::Result result) { prom->set_value(result); });
+
+        const Mission::Result result = future_result.get();
+        if (result != Mission::Result::Success) {
+            std::cout << "Mission upload failed (" << result << "), exiting." << std::endl;
+            return 1;
+        }
+        std::cout << "Mission uploaded." << std::endl;
+
+        // Now clear the mission!
+        mission.clear_mission_async(
+            [](Mission::Result callback) { std::cout << "Clear Mission Request" << std::endl; });
+        mission.subscribe_mission_progress([](Mission::MissionProgress progress) {
+            std::cout << "Current: " << progress.current << std::endl;
+            std::cout << "Total: " << progress.total << std::endl;
+        });
+    }
+
+    std::this_thread::sleep_for(std::chrono::seconds(20));
+
+    // We want to listen to the altitude of the drone at 1 Hz.
+    auto set_rate_result = telemetry.set_rate_position(1.0);
+    if (set_rate_result != mavsdk::Telemetry::Result::Success) {
+        std::cout << "Setting rate failed:" << set_rate_result << std::endl;
+        return 1;
+    }
+
+    // Set up callback to monitor altitude while the vehicle is in flight
+    telemetry.subscribe_position([](mavsdk::Telemetry::Position position) {
+        std::cout << "Altitude: " << position.relative_altitude_m << " m" << std::endl;
+    });
+
+    // Check if vehicle is ready to arm
+    while (telemetry.health_all_ok() != true) {
+        std::cout << "Vehicle is getting ready to arm" << std::endl;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    // Arm vehicle
+    std::cout << "Arming..." << std::endl;
+    const Action::Result arm_result = action.arm();
+
+    if (arm_result != Action::Result::Success) {
+        std::cout << ERROR_CONSOLE_TEXT << "Arming failed:" << arm_result << NORMAL_CONSOLE_TEXT
+                  << std::endl;
+        return 1;
+    }
+
+    action.set_takeoff_altitude(10.f);
+
+    // Take off
+    std::cout << "Taking off..." << std::endl;
+    bool takenOff = false;
+    while (true) {
+        const Action::Result takeoff_result = action.takeoff();
+        if (takeoff_result != Action::Result::Success) {
+            std::cout << ERROR_CONSOLE_TEXT << "Takeoff failed:" << takeoff_result
+                      << NORMAL_CONSOLE_TEXT << std::endl;
+            std::this_thread::sleep_for(std::chrono::seconds(1));
+            continue;
+        }
+        break;
+    }
+
+    while (true) {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
```

Comment:
I guess you don't have to join the threads because it just runs forever. Fair enough.